### PR TITLE
build: support containerized development via Docker; revert dash#4621

### PR DESCRIPTION
**Pull request by kittywhiskers**
## Why?

There is a possibility that a contributor's host system does not match the distribution used by the CI system that Dash uses, which makes debugging CI failures an increasing pain. Travis CI offered a feature where you could SSH into the CI instance and debug the cause of failure. GitLab CI, to my knowledge does not offer that

### CODE REVIEW

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
fix linter

File: contrib/containers/dev/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#
+# But if not, then you'll need to mount your system's root directory
+# (i.e. /) into the boot2docker instance if you want to mirror the exact
+# filesystem structure of your host.
+#
+    volumes:
+      - type: bind
+#        source: /host/$PWD # Workaround needed on non-Linux hosts
+        source: ../../..
+        target: /dash-src
```
Comment (UdjinM6) [inline contrib/containers/dev/docker-compose.yml:29]: nit:
```suggestion
        target: /dash-src

```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
needs rebase and a tiny fix, LGTM otherwise

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
ACK for merging via merge commit

**Pull request merged**